core/templates/components/forms/custom-forms-directives/audio-file-uploader.component.ts

```diff
@@ -24,19 +24,21 @@ import { downgradeComponent } from '@angular/upgrade/static';
   templateUrl: './audio-file-uploader.component.html'
 })
 export class AudioFileUploaderComponent {
-  @Input() droppedFile: FileList;
+  // Angular lifecycle hooks are used to populate values, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types.
+  @Input() droppedFile!: FileList;
   @Output() fileChange: EventEmitter<File> = new EventEmitter<File>();
   @Output() fileClear: EventEmitter<void> = new EventEmitter<void>();
-  @ViewChild('fileInput') fileInputRef: ElementRef;
-  @ViewChild('inputForm') inputFormRef: ElementRef;
+  // The following properties are used in function 'addAudio' which cannot
+  // be called until the input form and file input views are initialized.
+  @ViewChild('fileInput') fileInputRef!: ElementRef;
+  @ViewChild('inputForm') inputFormRef!: ElementRef;
   ALLOWED_AUDIO_FILE_TYPES = ['audio/mp3', 'audio/mpeg'];
-  errorMessage: string;
-
-  validateUploadedFile(file: File): string {
-    if (!file) {
```

Comment:
This condition is already being checked here:

https://github.com/oppia/oppia/blob/14d9f1e8b59fd6dd5c3ccea8fe4fb903379e20f9/core/templates/components/forms/custom-forms-directives/audio-file-uploader.component.ts#L59-L65

`file` will exist whenever `validateUploadedFile` is called

Reply:
Okay, but we should tackle this returned error message and place it in `addAudio` function maybe something like
```
 if (!file) {
      this.fileClear.emit();
      this.errorMessage = this.validateUploadedFile(file);
      return;
    }
```

Reply:
ummm, I don't think there is a need to display the message in the first place. The save button is **disabled** whenever the file is not uploaded. 

The way the `!file` condition reaches is when we have uploaded a file already and then we attempt to re-upload a file but cancel it.
Hence if we make the change you have suggested above then the following happens. 

https://user-images.githubusercontent.com/43031234/121644424-699cc700-caac-11eb-824b-24988121f0ac.mp4


I think it is more of a UX decision, having the save button disabled with the **no file chosen** displayed alongside `choose file` button and displaying the error message seems redundant, don't you think?

Reply:
Not sure, but this error message only display on clicking the cancel button and not for the first time, so I think this should be a right behaviour? @srijanreddy98 @vojtechjelinek wdyt?

Reply:
I am not quite sure about the context, which behavior should be the right one?

Reply:
Approving to unblock @srijanreddy98 @vojtechjelinek PTAL!

Reply:
I'm fine with both approaches, but I think uploading the file and then doing it again but clicking the cancel number is so rare case that I'm fine with removing the code.

Reply:
On the other hand, looking at this function separately I think the condition should be kept since `null` is not a valid file.

Reply:
@vojtechjelinek Take a look at the following code as stated above. 

https://github.com/oppia/oppia/blob/14d9f1e8b59fd6dd5c3ccea8fe4fb903379e20f9/core/templates/components/forms/custom-forms-directives/audio-file-uploader.component.ts#L59-L65

We cannot pass a `null` file in the parameter whenever it is called since the condition is already being checked.

Reply:
That doesn't really matter, but now I realize that the `null` cannot get there now since it will be typechecked. I'm fine with removing it then.

Reply:
alright.